### PR TITLE
Define a schema for the qhub configuration using pydantic

### PR DESCRIPTION
**Pull request by tonyfast**
This pr adds a schema and validation of a qhub configuration file.

- [x] pydantic schema
- [x] validation cli command
- [x] verify the schema
- [x] add dependencies
- [x] add tests

### CODE REVIEW

**Comment by costrouc** [comment]
@tonyfast could you also add https://github.com/Quansight/qhub-ops/blob/23e25753c7bce546afbc690fb6d6c9bc581db8d3/qhub_ops/cli/deploy.py#L18-L27 to the render cli command as well?

**Review by costrouc — CHANGES_REQUESTED** [review]
@tonyfast could you also add https://github.com/Quansight/qhub-ops/blob/23e25753c7bce546afbc690fb6d6c9bc581db8d3/qhub_ops/cli/deploy.py#L18-L27 to the render cli command as well https://github.com/Quansight/qhub-ops/blob/master/qhub_ops/cli/render.py#L15?

**Comment by tonyfast** [comment]
@costrouc is this what you're thinking?

File: qhub_ops/cli/render.py [hunk 1]
```diff
@@ -13,6 +14,7 @@ def create_render_subcommand(subparser):
 
 
 def handle_render(args):
+    verify(args.config)
```
Comment (costrouc) [inline qhub_ops/cli/render.py]: `args.config` is the path to the configuration file. Does pydantic handle this? I thought it expected a python object dict/list etc. .

**Review by costrouc — CHANGES_REQUESTED** [review]

File: qhub_ops/cli/render.py (hunk 1 above)
Comment (costrouc) [reply qhub_ops/cli/render.py]: I think adding should make it work

```python
config_filename = pathlib.Path(args.config) 
 if not config_filename.is_file(): 
     raise ValueError( 
         f"passed in configuration filename={config_filename} must exist" 
     ) 
  
 with config_filename.open() as f: 
     config = yaml.safe_load(f.read()) 
```

Comment (tonyfast) [reply qhub_ops/cli/render.py]: makes sense. i wasnt sure what was coming the config object.

**Review by costrouc — APPROVED** [review]

**Comment by costrouc** [comment]
Looks good to me! Thanks @tonyfast

**Pull request merged**